internal/index_read.go

```diff
@@ -102,26 +103,19 @@ func readBins(r io.Reader, typ string) ([]Bin, *ReferenceStats, error) {
 	return bins, stats, nil
 }
 
-func readChunks(r io.Reader, n int32, typ string) ([]bgzf.Chunk, error) {
+func readChunks(r io.Reader, n int32, typ string, buf [16]byte) ([]bgzf.Chunk, error) {
```

Comment:
I did that first but the loop is usually only run between 1 and 3 times per function call so the benefit was minor. This gave a little bit better improvement, though most of the change improvement comes from the linear index.